clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/deploy/ops/actions/PrepareUpdateAutoScalingGroup.java

```diff
@@ -0,0 +1,102 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.aws.deploy.ops.actions;
+
+import com.amazonaws.services.autoscaling.model.LaunchTemplateOverrides;
+import com.amazonaws.services.ec2.model.LaunchTemplateVersion;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.BasicAmazonDeployDescription.LaunchTemplateOverridesForInstanceType;
+import com.netflix.spinnaker.clouddriver.aws.deploy.description.ModifyServerGroupLaunchTemplateDescription;
+import com.netflix.spinnaker.clouddriver.event.EventMetadata;
+import com.netflix.spinnaker.clouddriver.saga.SagaCommand;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.models.Saga;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.stereotype.Component;
+
+/** Action to prepare for AWS EC2 Auto Scaling Group update. */
+@Component
+public class PrepareUpdateAutoScalingGroup
+    implements SagaAction<PrepareUpdateAutoScalingGroup.PrepareUpdateAutoScalingGroupCommand> {
+  @NotNull
+  @Override
+  public Result apply(@NotNull PrepareUpdateAutoScalingGroupCommand command, @NotNull Saga saga) {
+    ModifyServerGroupLaunchTemplateDescription description = command.description;
+
+    saga.log(
+        "[SAGA_ACTION] Preparing to update EC2 Auto Scaling Group " + description.getAsgName());
+
+    // transform overrides
+    List<LaunchTemplateOverridesForInstanceType> overridesInReq =
+        description.getLaunchTemplateOverridesForInstanceType();
+    List<LaunchTemplateOverrides> ltOverrides = null;
+    if (overridesInReq != null && !overridesInReq.isEmpty()) {
+      ltOverrides =
+          description.getLaunchTemplateOverridesForInstanceType().stream()
+              .map(
+                  o ->
+                      new LaunchTemplateOverrides()
+                          .withInstanceType(o.getInstanceType())
+                          .withWeightedCapacity(o.getWeightedCapacity()))
+              .collect(Collectors.toList());
+    }
+
+    UpdateAutoScalingGroup.UpdateAutoScalingGroupCommand updateCommand =
+        UpdateAutoScalingGroup.UpdateAutoScalingGroupCommand.builder()
+            .description(description)
+            .launchTemplateVersion(command.launchTemplateVersion)
+            .launchTemplateOverrides(ltOverrides)
```

Comment:
Does this take nullable values? `ltOverrides` would be `null` if not set in the condition.

Reply:
Yes, in L#57, `ltOverrides` is null. If there are no overrides and `ltOverrides` remains null at this point, it will be simply passed over as `UpdateAutoScalingGroupCommand` specifies it as a nullable field in UpdateAutoScalingGroup.java L#231